pulsar-client-cpp/lib/ConsumerImpl.cc

```diff
@@ -879,4 +883,9 @@ void ConsumerImpl::brokerConsumerStatsListener(Result res, BrokerConsumerStatsIm
     }
 }
 
+bool ConsumerImpl::isReadCompacted() {
+    Lock lock(mutex_);
```

Comment:
Why the lock at all? readCompacted_ can be const, and initialized in the initialization list

Reply:
will remove it.